common/assets/asset-downloader.nix

```diff
@@ -0,0 +1,78 @@
+{ pkgs, config, lib, ... }:
+let
+  getAsset = { url, name, sha256 }:
+    pkgs.stdenvNoCC.mkDerivation {
+      name = "asset-${name}";
+      meta.description = "asset: ${name}";
+      src = pkgs.fetchurl { inherit url sha256; };
+      buildInputs = [ pkgs.coreutils ];
+      phases = [ "unpackPhase" "installPhase" ];
+
+      unpackPhase = "true";
+
+      installPhase = ''
+        mkdir -p "$(dirname $out/"${name}")"
+        cp -vf $src $out/"${name}"
+      '';
+    };
+
+  getAssets = assets:
+    pkgs.symlinkJoin {
+      name = "assets";
+      paths = map getAsset assets;
+    };
+
+  assetsPkg = getAssets config.mixos.assets;
+
+  symlink-assets-at-home = pkgs.writeShellApplication {
```

Comment:
can we make this do the symlinks with home-manager or it is going to be too complicated?

Reply:
yes, thanks! it is much cleaner this way